File: 6/src6/weeks/3/sigma1.py
# Sums a range of numbers recursively


from cs50 import get_int


def main():
    while True:
        n = get_int("Positive integer: ")
        if n > 0:
            break
    answer = sigma(n)
    print(answer)


def sigma(m):
    """Return sum of 1 through m"""
    if m <= 0:
        return 0
    else:
        return m + sigma(m - 1)


if __name__ == "__main__":
    main()

File: 3/src3/sigma1.c
// Sums a range of numbers recursively

#include <cs50.h>
#include <stdio.h>

int sigma(int m);

int main(void)
{
    int n;
    do
    {
        n = get_int("Positive integer: ");
    }
    while (n < 1);
    int answer = sigma(n);
    printf("%i\n", answer);
}

// Returns sum of 1 through m
int sigma(int m)
{
    if (m <= 0)
    {
        return 0;
    }
    else
    {
        return (m + sigma(m - 1));
    }
}
